### PR TITLE
Support LibreOffice 5.4.6.2 on Mac OS X 10.13.3 with case-sensitive filesystem

### PR DESCRIPTION
**Pull request by amotl**
Hi there,

first things first: Thanks for conceiving and maintaining `unoconv`, it saves us a lot of troubles when talking to LibreOffice and is a real pleasure to use.

This PR improves the situation regarding appropriately finding the LibreOffice assets along the lines of #27, #49, #263, #391 and #442. We are running LibreOffice 5.4.6.2 on Mac OS X 10.13.3 with a case-sensitive filesystem, `unoconv` was installed using `brew install unoconv`.

These amendments account for:
- Use `/Applications/LibreOffice.app/Contents/Resources/python` as yet another candidate for finding the `python` executable shipped with LibreOffice on Mac OS X
- Support case-sensitive filesystems on Mac OS X re. capitalized notation of the `Resources` folder
- Fix manipulation of `os.environ['PATH']`
- Report error if `os.spawnvpe()` fails

We hope you like this.

With kind regards,
Andreas.

### CODE REVIEW

**Comment by amotl** [comment]
As this once passed for LibreOffice version 4.2.8.2 and also once for 5.4.3.2 as well as 5.4.6.2 (all on Mac OS X), i just assume Travis flakeyness. Everything looks good on Linux, so there seem to be no regressions.

The Travis builds in question are:
- https://travis-ci.org/dagwieers/unoconv/builds/369400272
- https://travis-ci.org/dagwieers/unoconv/builds/369402844

**Comment by Eitz** [comment]
We need this ASAP.

File: unoconv [hunk 1]
```diff
@@ -197,23 +197,47 @@ def find_offices():
...
+        ret.append(office)
+
     return ret
 
+def find_file_candidate(folders, filenames):
+    for folder in folders:
+        for filename in filenames:
+            candidate = realpath(folder, filename)
+            if os.path.isfile(candidate):
+                return candidate
```
Comment (regebro) [inline unoconv]: Shouldn't this function return a list of all that are found, not just one?

File: unoconv [hunk 2]
```diff
@@ -197,23 +197,47 @@ def find_offices():
...
 
-        for pythonbinary in pythonbinaries:
-            if os.path.isfile(realpath(unopath, pythonbinary)):
-                info(3, "Found %s in %s" % (pythonbinary, unopath))
-                ret.append(Office(basepath, urepath, unopath, officelibrary, officebinary,
-                                  realpath(unopath, pythonbinary), pythonhome))
+
+        ### Find suitable Python executable
+        pythonpath_candidates = [
+
```
[the remark this answers is not in the document]
Comment (amotl) [reply unoconv]: Will be happy to compress this. To be sure: You are referring to the blank lines inside the definition of `pythonpath_candidates`?

File: unoconv (hunk 1 above)
Comment (amotl) [reply unoconv]: In this case, i really wanted the helper function to return the single outcome from a list of possible candidates. I wouldn't know what to do with multiple outcomes on the caller side. Who would decide which one to use?

So, maybe we should rename the function to `find_file_effective` or `find_file_from_candidates` then. Would that help?

File: unoconv (hunk 2 above)
[the remark this answers is not in the document]
Comment (regebro) [reply unoconv]: yes

File: unoconv (hunk 1 above)
Comment (regebro) [reply unoconv]: `find_python` or `find_executable` seems better. "file_effective" doesn't explain what file or why it's effective.

Comment (regebro) [reply unoconv]: Other than that it looks good.

File: unoconv (hunk 2 above)
[the remark this answers is not in the document]
Comment (amotl) [reply unoconv]: Thanks, resolved by corresponding commit down the line.

File: unoconv (hunk 1 above)
Comment (amotl) [reply unoconv]: It's called `find_executable` now. Hope this helps and thanks again.

**Comment by amotl** [comment]
I've implemented your requests from code review after rebasing on top of master. The amendments have been squashed and force-pushed.

**Comment by amotl** [comment]
Thanks for merging, @regebro!

**Pull request merged**